src/SlackEventHandler.ts

```diff
@@ -228,29 +234,23 @@ export class SlackEventHandler extends BaseSlackHandler {
             // Filter out tombstones early, we only care about them on deletion.
             throw Error("ignored");
         }
+
         // Only count received messages that aren't self-reflections
         this.main.incCounter(METRIC_RECEIVED_MESSAGE, {side: "remote"});
 
         if (event.type === "channel_join") {
-            await room.onSlackUserJoin(event.user!, event.inviter!);
+            await room.onSlackUserJoin(userOrBotId!, event.inviter!);
         } else if (event.type === "channel_leave") {
-            await room.onSlackUserLeft(event.user!);
+            await room.onSlackUserLeft(userOrBotId!);
         }
 
-        const msg = Object.assign({}, event, {
+        const msg = {
+            ...event,
             channel_id: event.channel,
             team_domain: team.domain || team.id,
             team_id: teamId,
-            user_id: event.user || event.bot_id,
-        });
-
-        if (event.type === "reaction_added") {
-            return room.onSlackReactionAdded(msg, teamId);
-        }
-        // TODO: We cannot remove reactions yet, see https://github.com/matrix-org/matrix-appservice-slack/issues/154
-        /* else if (params.event.type === "reaction_removed") {
-            return room.onSlackReactionRemoved(msg);
-        } */
+            user_id: userOrBotId!,
```

Comment:
Ah, I see why you created `userOrBotId`. I'm not sure if we should pass this value to `onSlackUserJoin|Left` though.